File: Solutions/C++/704. Binary Search.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int index = lower_bound(nums.begin(),nums.end(),target)-nums.begin();
        if(index < nums.size())
            if(nums[index] == target)
            return index;
        return -1;
    }
    
};

File: Solutions/python/704-Binary-Search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        low=0
        high=len(nums)-1
        while low<=high:
            mid=int((low+high)/2)
            if(nums[mid]==target):
                return mid 
            elif(nums[mid]<target):
                low=mid+1
            elif(nums[mid]>target):
                high=mid-1
        return -1
